Gems/Atom/RPI/Code/Source/RPI.Public/WindowContext.cpp

```diff
@@ -68,27 +71,33 @@ namespace AZ
 
         const RHI::Ptr<RHI::SwapChain>& WindowContext::GetSwapChain(SwapChainMode swapChainMode) const
         {
-            return m_swapChains[static_cast<uint32_t>(swapChainMode)];
+            uint32_t swapChainIndex = static_cast<uint32_t>(swapChainMode);
+            AZ_Assert(swapChainIndex < GetSwapChainsSize(), "Swapchain does not exist");
```

Comment:
It'd be informative to print the index and size in the assert message. Same in the other asserts.